gemrb/core/GUI/View.cpp

```diff
@@ -141,6 +141,14 @@ void View::InvalidateSubviews(const Region& rgn) const
 	}
 }
 
+void View::SetVisible(bool vis) {
+	SetFlags(Invisible, vis ? BitOp::NAND : BitOp::OR);
+
+	if (!vis && superView != nullptr) {
```

Comment:
I think we need to do this regardless of `vis`. I cant think of a real world scenario, but if there were some case where a superview was tainted in the same region (eg overlapping view, or swapping a view with another which we do) as the view becoming visible, and the now visible view wasn't fully opaque, there could be artifacts.

I dont think there is much payoff optimizing it compared to difficulty debugging such a scenario.

Reply:
Alright